modules/imgproc/src/hough.cpp

```diff
@@ -1002,222 +1006,770 @@ cvHoughLines2( CvArr* src_image, void* lineStorage, int method,
 *                                     Circle Detection                                   *
 \****************************************************************************************/
 
-static void
-icvHoughCirclesGradient( CvMat* img, float dp, float min_dist,
-                         int min_radius, int max_radius,
-                         int canny_threshold, int acc_threshold,
-                         CvSeq* circles, int circles_max )
+#if CV_SIMD128
+static
+inline unsigned int trailingZeros(unsigned int value) {
```

Comment:
We have the same function in contours.cpp.
It is very generic function, so it is better to move these implementations into [opencv2/core/hal/intrin.hpp](https://github.com/opencv/opencv/blob/master/modules/core/include/opencv2/core/hal/intrin.hpp) file (at the end of file).
Probably under this signature (and without `#if CV_SIMD128`):
```
inline unsigned int trailingZeros32(unsigned int value)
```